src/pymedphys/_labs/paulking/profile.py

```diff
@@ -81,42 +110,40 @@ def interactive(self):
 
 
 class DoseProfile(Dose1D):
-    def __init__(self, *args, metadata={}, x=[], data=[], **kwargs):
-        try:  # DoseProfile( x=[0,2,4], data=[1,3,5] )
-            x = kwargs.pop('x')
-            data = kwargs.pop('data')
-        except (KeyError):
-            pass
-
-        try:  # DoseProfile( [0,2,4], [1,3,5] )
-            assert (len(args) == 2)
-            assert len(args[0]) == len(args[1])
-            x = args[0]
-            data = args[1]
-        except (AssertionError, IndexError):
-            pass
-
-        try:  # DoseProfile( [(0,1),(2,3),(4,5) ]
-            assert len(args) == 1
-            assert len(args[0][0]) == 2
-            x = list(list(zip(*args[0]))[0])
-            data = list(list(zip(*args[0]))[1])
-        except (AssertionError, IndexError):
-            pass
-
-        super().__init__(x, data)
-
+    def __init__(self, x, data, metadata={}):
         self.metadata = metadata
+        super().__init__(x, data)
 
     def __len__(self):
         assert len(self.x) == len(self.data)
         return len(self.x)
 
+    def from_tuples(self, list_of_tuples, metadata={}):
+        """ Load a list of (x,data) tuples.
+
+        Overwrites any existing dose profile data and metadata.
+
+        Arguments
+        ---------
+        list_of_tuples : list
+            List of (float x, float data) tuples.
+
+        Keyword Arguments
+        -----------------
+        metadata : dict, optional
+            Dictionary of key-value pairs that describe the profile
+
+        """
+        self.metadata = metadata
+        x = list(list(zip(*list_of_tuples))[0])
+        data = list(list(zip(*list_of_tuples))[1])
+        super().__init__(x, data)
+
     def interactive(self):
-        pass
+        pass  # WHAT IS THIS INTENDED TO DO?
```

Comment:
My thought was it would be for an interactive plotting function using bokeh.

Reply:
Fair enough. Though I don't know boken.